File: application/backend-credit-card-enrollment/backend-java/src/main/java/cloud/ambar/creditcard/enrollment/event/EnrollmentRequested.java
package cloud.ambar.creditcard.enrollment.event;

import cloud.ambar.common.event.CreationEvent;
import cloud.ambar.creditcard.enrollment.aggregate.Enrollment;
import cloud.ambar.creditcard.enrollment.aggregate.EnrollmentStatus;
import lombok.Getter;
import lombok.NonNull;
import lombok.experimental.SuperBuilder;

@SuperBuilder
@Getter
public class EnrollmentRequested extends CreationEvent<Enrollment> {
    @NonNull private String userId;
    @NonNull private String productId;
    @NonNull private Integer annualIncomeInCents;

    public Enrollment createAggregate() {
        return Enrollment.builder()
                .aggregateId(aggregateId)
                .aggregateVersion(aggregateVersion)
                .userId(userId)
                .productId(productId)
                .enrollmentFirstRequestedOn(recordedOn)
                .annualIncomeInCents(annualIncomeInCents)
                .status(EnrollmentStatus.Requested.name())
                .build();
    }
}

File: application/backend-credit-card-enrollment/backend-c#/CreditCard/Enrollment/Event/EnrollmentRequested.cs
using CreditCardEnrollment.Common.Event;

namespace CreditCardEnrollment.CreditCard.Enrollment.Event;

public class EnrollmentRequested : CreationEvent<Aggregate.Enrollment>
{
    public required string UserId { get; init; }
    public required string ProductId { get; init; }
    public required int AnnualIncomeInCents { get; init; }

    public override Aggregate.Enrollment CreateAggregate()
    {
        return new Aggregate.Enrollment
        {
            AggregateId = AggregateId,
            AggregateVersion = AggregateVersion,
            UserId = UserId,
            ProductId = ProductId,
            Status = Aggregate.EnrollmentStatus.Requested,
            AnnualIncomeInCents = AnnualIncomeInCents,
            EnrollmentFirstRequestedOn = RecordedOn
        };
    }
}
